Microsoft.Research/Contracts/WindowsBase/System.ComponentModel.ICollectionView.cs

```diff
@@ -0,0 +1,505 @@
+        //     an item within the view; otherwise, false.
+        bool MoveCurrentToPosition(int position);
+        //
+        // Summary:
+        //     Sets the item before the System.ComponentModel.ICollectionView.CurrentItem
+        //     in the view as the System.ComponentModel.ICollectionView.CurrentItem.
+        //
+        // Returns:
+        //     true if the resulting System.ComponentModel.ICollectionView.CurrentItem is
+        //     an item within the view; otherwise, false.
+        bool MoveCurrentToPrevious();
+        //
+        // Summary:
+        //     Recreates the view.
+        void Refresh();
+    }
+
+    [ContractClassFor(typeof (ICollectionView))]
+    abstract class ICollectionViewContract : ICollectionView
+    {
+        event NotifyCollectionChangedEventHandler INotifyCollectionChanged.CollectionChanged
+        {
+            add
+            {
+            }
+            remove
+            {
+            }
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            throw new NotImplementedException();
+        }
+
+        bool ICollectionView.CanFilter
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        bool ICollectionView.CanGroup
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        bool ICollectionView.CanSort
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        CultureInfo ICollectionView.Culture
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+            set
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        object ICollectionView.CurrentItem
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        int ICollectionView.CurrentPosition
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        Predicate<object> ICollectionView.Filter
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+            set
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        ObservableCollection<GroupDescription> ICollectionView.GroupDescriptions
+        {
+            get
+            {
+                Contract.Ensures(Contract.Result<ObservableCollection<GroupDescription>>() != null);
+                throw new NotImplementedException();
+            }
+        }
+
+        ReadOnlyObservableCollection<object> ICollectionView.Groups
+        {
+            get
+            {
+                Contract.Ensures(Contract.Result<ReadOnlyObservableCollection<object>>() != null);
```

Comment:
Then this documentation might be wrong. Generally plain properties returning ObservableCollections must never change the returned object, since you need to connect to the returned objects CollectionChanged event to wait for some object to arrive - which would be impossible if null is returned. 
Also since this is a read only property, we could not set a new collection but only add items to the existing collection - which is also impossible if null would be returned.